pkg/controller/storagecluster/reconcile.go

```diff
@@ -1102,8 +1115,19 @@ func (r *ReconcileStorageCluster) deleteResources(sc *ocsv1.StorageCluster, reqL
 //getCephDaemonPlacements returns placement configuration for ceph components with appropriate topology
 func getCephDaemonPlacements(sc *ocsv1.StorageCluster, component string) rook.Placement {
 	placement := rook.Placement{}
-	in := defaults.DaemonPlacements[component]
-	(&in).DeepCopyInto(&placement)
+	in, ok := sc.Spec.Placement[rook.KeyType(component)]
```

Comment:
Don't use `rook.KeyType()` here. We're not validating to the level Rook is, and we have different requirements than Rook-Ceph does.

Reply:
It is required for a type of  ```rook.Placement``` and using any string works.